### PR TITLE
Fix to allow re-creation of volume, which was deleted out of band, with the same name and by the same volume driver

### PR DESCRIPTION
**Pull request by ssuvasanth**
PR #27164 is perfectly fine when purging the stale volume name and allowing the creation of a volume with the same name when it is created by a different volume driver than the one used to create it earlier.

But it is desirable to let the same volume driver create the volume again which was deleted out of the band.

So, if the volume does not exist, purge it from the cache and allow the re-creation by any driver. 

On the other hand, if the volume exists and a create is requested,it is more appropriate to get the response from the volume driver as the volume is totally controlled by the volume driver and docker daemon has no control over managing that volume.

I have verified the behaviour.

### CODE REVIEW

File: volume/store/store.go [hunk 1]
```diff
@@ -237,6 +238,72 @@ func (s *VolumeStore) Create(name, driverName string, opts, labels map[string]st
...
+// particularly around locking the store.
+// TODO(cpuguy83): With v2 plugins this shouldn't be a problem. Could also potentially
+// use a connect timeout for this kind of check to ensure we aren't blocking for a
+// long time.
+func (s *VolumeStore) checkConflict(name, driverName string) (volume.Volume, error) {
+	// check the local cache
+	v, _ := s.getNamed(name)
+	if v != nil {
+		vDriverName := v.DriverName()
+		if driverName != "" {
```
Comment (AkihiroSuda) [inline volume/store/store.go]: Probably vDriverName?

Comment (ssuvasanth) [inline volume/store/store.go]: 'driverName' looks fine to me. vDriverName is the driver that was used to create the volume that is present in the cache but 'driverName' is the new driver that is being used to create the volume again. This creation succeeds if the volume does not exist even if it is present in the cache.
driverName is passed as a parameter to the function.

Comment (ssuvasanth) [inline volume/store/store.go]: @AkihiroSuda  does this need any change?

Comment (ssuvasanth) [inline volume/store/store.go]: @AkihiroSuda Thank you for your feedback. Appreciate it. I have made the changes.

Appreciate your time if you could review again.

**Comment by ssuvasanth** [comment]
Closing this with a better approach in PR #27274.

**Pull request closed without merging**